src/modules/projectConfig/projectConfig.ts

```diff
@@ -94,48 +72,8 @@ export class ProjectConfig {
         }
         projectConfig.getVariableMappings().set('appManifestPath', DEFAULT_APP_MANIFEST_PATH);
         projectConfig.getVariableMappings().set('githubRepoId', '<myrepo>');
-        projectConfig.write();
-    }
-
-    /**
-     * Creates a string out of the project configuration in format for the lock file.
-     * @returns A string for writing the ProjectConfigLock.
-     */
-    toLockString(): string {
-        const packagesObject: { [key: string]: string } = {};
-        this.getPackages().forEach((packageConfig: PackageConfig) => {
-            packagesObject[`${packageConfig.repo}`] = packageConfig.version;
-        });
-        const projectConfigAttributes = {
-            packages: packagesObject,
-        };
-
-        const configString = `${JSON.stringify(projectConfigAttributes, null, 4)}\n`;
-        return configString;
-    }
-
-    /**
-     * Write the project configuration to file.
-     *
-     * @param path Path of the file to write the configuration to.
-     */
-    write(path: PathLike = DEFAULT_CONFIG_FILE_PATH): void {
-        // if we find an "old" project configuration with no components explicitly mentioned
-        // we persist all components we can find.
-        let componentsToSerialize: ComponentConfig[] = this._components;
-
-        if (!componentsToSerialize || componentsToSerialize.length === 0) {
-            componentsToSerialize = this.getComponents().map((cc) => cc.config);
-        }
-
-        const projectConfigAttributes = {
-            packages: ConfigFileParser.toWritablePackageConfig(this._packages),
-            components: ConfigFileParser.toWritableComponentConfig(componentsToSerialize),
-            variables: this._variables,
-            cliVersion: this.cliVersion,
-        };
-        const configString = `${JSON.stringify(projectConfigAttributes, mapReplacer, 4)}\n`;
-        CliFileSystem.writeFileSync(path, configString);
+        const projectConfigWriter = new ProjectConfigWriter();
```

Comment:
ProjectConfig::create() should not create a writer but return a ProjectConfig instance instead which can then be written to a file by the caller using a writer.